valhalla/userrequests/state_changes.py

```diff
@@ -233,14 +233,15 @@ def update_request_states_for_window_expiration():
 
 def update_request_states_from_pond_blocks(pond_blocks):
     '''Update the states of requests and user_requests given a set of recently changed pond blocks.'''
-    sorted_blocks = sorted(pond_blocks, key=lambda x: x['molecules'][0]['tracking_number'])
-    blocks_by_tracking_num = itertools.groupby(sorted_blocks, lambda x: x['molecules'][0]['tracking_number'])
+    blocks_with_tracking_nums = [pb for pb in pond_blocks if pb['molecules'][0]['tracking_num']]
```

Comment:
Might want to use .get() here, if you aren't confident a block will have a tracking_num